lib/core/src/display/symbol/geometry/sprite.rs

```diff
@@ -0,0 +1,288 @@
+        t.display_object.clone_ref()
+    }
+}
+
+impl<'t> Modify<&'t DisplayObjectData> for &'t SpriteData {
+    fn modify<F:FnOnce(&'t DisplayObjectData)>(self, f:F) {
+        f(&self.display_object)
+    }
+}
+
+impl Drop for SpriteData {
+    fn drop(&mut self) {
+        self.sprite_ref.symbol_ref.world.mod_stats(|stats| stats.dec_sprite_count());
+
+        let mut world = self.sprite_ref.symbol_ref.world.borrow_mut();
+        let symbol    = &mut world.workspace[self.sprite_ref.symbol_ref.symbol_id];
+        let mesh      = &mut symbol.surface;
+        self.bbox.set(Vector2::new(0.0,0.0));
+        mesh.scopes.instance.dispose(self.sprite_ref.instance_id);
+        self.display_object.unset_parent();
+    }
+}
+
+
+
+// ====================
+// === SpriteSystem ===
+// ====================
+
+/// Creates a set of sprites. All sprites in the sprite system share the same material. Sprite
+/// system is a very efficient way to display geometry. Sprites are rendered as instances of the
+/// same mesh. Each sprite can be controlled by the instance and global attributes.
+pub struct SpriteSystem {
+    display_object    : DisplayObjectData,
+    symbol_ref        : SymbolRef,
+    transform         : Buffer<Matrix4<f32>>,
+    _uv               : Buffer<Vector2<f32>>,
+    bbox              : Buffer<Vector2<f32>>,
+}
+
+impl SpriteSystem {
+    /// Constructor.
+    pub fn new(world:&World) -> Self {
+        let logger         = Logger::new("SpriteSystem");
+        let display_object = DisplayObjectData::new(logger);
+        let world_data     = &mut world.borrow_mut();
+        let workspace      = &mut world_data.workspace;
+        let symbol_id      = workspace.new_symbol();
+        let symbol         = &mut workspace[symbol_id];
+        let mesh           = &mut symbol.surface;
+        let uv             = mesh.scopes.point.add_buffer("uv");
+        let transform      = mesh.scopes.instance.add_buffer("transform");
+        let bbox           = mesh.scopes.instance.add_buffer("bounds");
+
+        let geometry_material = Self::geometry_material();
+        let material          = Self::material();
+
+        symbol.shader.set_geometry_material (&geometry_material);
+        symbol.shader.set_material          (&material);
+
+        let p1_index = mesh.scopes.point.add_instance();
+        let p2_index = mesh.scopes.point.add_instance();
+        let p3_index = mesh.scopes.point.add_instance();
+        let p4_index = mesh.scopes.point.add_instance();
+
+        uv.get(p1_index).set(Vector2::new(0.0, 0.0));
+        uv.get(p2_index).set(Vector2::new(0.0, 1.0));
+        uv.get(p3_index).set(Vector2::new(1.0, 0.0));
+        uv.get(p4_index).set(Vector2::new(1.0, 1.0));
+
+        world_data.stats.inc_sprite_system_count();
+
+        let world      = world.clone_ref();
+        let symbol_ref = SymbolRef::new(world,symbol_id);
+        Self {display_object,symbol_ref,transform,_uv:uv,bbox}
+    }
```

Comment:
Long function, maybe worth splitting somehow.

Reply:
TBH, I was thinking about it but dont know how to do it propelry. All the lines are so interconnected. Any recommendations here?